src/marker.rs

```diff
@@ -634,14 +634,15 @@ impl<'py> Python<'py> {
         locals: Option<&Bound<'py, PyDict>>,
     ) -> PyResult<Bound<'py, PyAny>> {
         unsafe {
-            let mptr = ffi::PyImport_AddModule(ffi::c_str!("__main__").as_ptr());
+            let mptr = ffi::compat::PyImport_AddModuleRef(ffi::c_str!("__main__").as_ptr());
```

Comment:
It'd be better to use `Bound` here (same for the other raw pointers in this function...);

Reply:
👍 on avoiding the manual reference counting. This can change to 

```suggestion
            let mptr = ffi::compat::PyImport_AddModuleRef(ffi::c_str!("__main__").as_ptr()).assume_owned_or_err(py)?;
```

It looks like `PyModule_GetDict` below returns a borrowed reference. Is that problematic under freethreading? I assume so, do we need to call and incref under a critical section? It doesn't look like there is a `PyModule_GetDictRef`. Should we just do `.getattr("__dict__")`?

Similarly it looks like `PyEval_GetBuiltins` is deprecated in 3.13, so I guess we need to update that in a follow up...

Reply:
Good call about `PyModule_GetDict`. I wonder if we don't actually need that on free-threaded builds since we don't need to update `globals` on Python 3.10 and newer. I'll see if we can avoid it with conditional compilation. 

I think we could also probably use [`PyModule_AddObjectRef`](https://docs.python.org/3/c-api/module.html#c.PyModule_AddObjectRef) or a compat definition based on `PyModule_AddObject` instead.